src/lib/firewall.ts

```diff
@@ -137,56 +137,65 @@ async function applyFirewall(
-		mode = 'off';
-	}
+	try {
+		// only apply valid mode...
+		if (!ALLOWED_MODES.includes(mode)) {
+			log.warn(`Invalid firewall mode: ${mode}. Reverting to state: off`);
+			mode = 'off';
+		}
 
-	log.info(`${LOG_PREFIX} Applying firewall mode: ${mode}`);
-
-	// get an adaptor to manipulate iptables rules...
-	const ruleAdaptor = iptables.getDefaultRuleAdaptor();
-
-	// are we running services in host-network mode?
-	const isServicesInHostNetworkMode = await runningHostBoundServices();
-
-	// should we allow only traffic to the balena host services?
-	const returnIfOff: iptables.Rule | iptables.Rule[] =
-		mode === 'off' || (mode === 'auto' && !isServicesInHostNetworkMode)
-			? {
-					comment: `Firewall disabled (${mode})`,
-					action: iptables.RuleAction.Insert,
-					target: 'RETURN',
-			  }
-			: [];
-
-	// configure the BALENA-FIREWALL chain...
-	await iptables
-		.build()
-		.forTable('filter', (filter) =>
-			filter
-				.forChain(BALENA_FIREWALL_CHAIN, (chain) =>
-					chain
-						.addRule(prepareChain)
-						.addRule(supervisorAccessRules)
-						.addRule(standardServices)
-						.addRule(standardPolicy)
-						.addRule(returnIfOff),
-				)
-				.forChain('INPUT', (chain) =>
-					chain
-						.addRule({
-							action: iptables.RuleAction.Flush,
-						})
-						.addRule({
-							action: iptables.RuleAction.Append,
-							target: 'BALENA-FIREWALL',
-						}),
-				),
-		)
-		.apply(ruleAdaptor);
-
-	// all done!
-	log.success(`${LOG_PREFIX} Firewall mode applied`);
+		log.info(`${LOG_PREFIX} Applying firewall mode: ${mode}`);
+
+		// get an adaptor to manipulate iptables rules...
+		const ruleAdaptor = iptables.getDefaultRuleAdaptor();
+
+		// are we running services in host-network mode?
+		const isServicesInHostNetworkMode = await runningHostBoundServices();
+
+		// should we allow only traffic to the balena host services?
+		const returnIfOff: iptables.Rule | iptables.Rule[] =
+			mode === 'off' || (mode === 'auto' && !isServicesInHostNetworkMode)
+				? {
+						comment: `Firewall disabled (${mode})`,
+						action: iptables.RuleAction.Insert,
+						target: 'RETURN',
+				  }
+				: [];
+
+		// configure the BALENA-FIREWALL chain...
+		await iptables
+			.build()
+			.forTable('filter', (filter) =>
+				filter
+					.forChain(BALENA_FIREWALL_CHAIN, (chain) =>
+						chain
+							.addRule(prepareChain)
+							.addRule(supervisorAccessRules)
+							.addRule(standardServices)
+							.addRule(standardPolicy)
+							.addRule(returnIfOff),
+					)
+					.forChain('INPUT', (chain) =>
+						chain
+							.addRule({
+								action: iptables.RuleAction.Flush,
+							})
+							.addRule({
+								action: iptables.RuleAction.Append,
+								target: 'BALENA-FIREWALL',
+							}),
+					),
+			)
+			.apply(ruleAdaptor);
+
+		// all done!
+		log.success(`${LOG_PREFIX} Firewall mode applied`);
+	} catch (err) {
+		if (err instanceof iptables.IPTablesRuleError) {
+			log.debug(`Ruleset:\r\n${err.ruleset}`);
```

Comment:
It might make more sense to add this after the other error printouts. Also it's probably worth the user knowing on the dashboard that the firewall failed to apply.